pkg/apis/scheduling/validation/validation_test.go

```diff
@@ -0,0 +1,150 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package validation
+
+import (
+	"testing"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+	"k8s.io/kubernetes/pkg/apis/scheduling"
+)
+
+func TestValidatePriorityClass(t *testing.T) {
+	successCases := map[string]scheduling.PriorityClass{
+		"no description": {
+			ObjectMeta: metav1.ObjectMeta{Name: "tier1", Namespace: ""},
+			Value:      100,
+		},
+		"with description": {
+			ObjectMeta:    metav1.ObjectMeta{Name: "tier1", Namespace: ""},
+			Value:         200,
+			GlobalDefault: false,
+			Description:   "Used for the highest priority pods.",
+		},
+	}
+
+	for k, v := range successCases {
+		if errs := ValidatePriorityClass(&v); len(errs) != 0 {
+			t.Errorf("Expected success for %s, got %v", k, errs)
+		}
+	}
+
+	errorCases := map[string]scheduling.PriorityClass{
+		"with namespace": {
+			ObjectMeta: metav1.ObjectMeta{Name: "tier1", Namespace: "foo"},
+			Value:      100,
+		},
+		"invalid name": {
+			ObjectMeta: metav1.ObjectMeta{Name: "tier&1", Namespace: ""},
+			Value:      100,
+		},
+	}
+
+	for k, v := range errorCases {
+		if errs := ValidatePriorityClass(&v); len(errs) == 0 {
+			t.Errorf("Expected error for %s, but it succeeded", k)
+		}
+	}
+}
+
+func TestValidatePriorityClassUpdate(t *testing.T) {
+	old := scheduling.PriorityClass{
+		ObjectMeta: metav1.ObjectMeta{Name: "tier1", Namespace: "", ResourceVersion: "1",
+			Annotations: map[string]string{
+				"kbernetes.io/description": "Used for the highest priority pods.",
```

Comment:
Actually, I removed the whole annotation. I learned that we should not use annotations for information which should be discoverable by users. So, I changed description from an annotation to a field and this was a remnant of the initial code. Thanks for pointing it out!